### DISCUSSION TITLE
监听delete的时候怎么获得要删除的节点的id呢

### DISCUSSION THREAD

**Discussion by MR1499** [hizzgdev/jsmind#527] [General]
你好
document.addEventListener('keydown', function(e) {
    if (e.code === 'Delete' || e.key === 'Delete') {  // 兼容性检查
        var selected_node = jm.get_selected_node();
        // alert(selected_node.id)
        // 删除逻辑
        }
});
我在监听Delete进行节点删除的时候发现，直接这么写会导致删除删掉我想删除的节点的父节点
原因可能是，jsmind里已经监听了delete按键的事件，并且会删掉对应的节点，删掉后会选中父节点
而当我再监听的时候就会获得此时的nodeid，但是，这时候实际是我想删除的节点的父节点的nodeid
能不能把删掉的节点的id做一个返回值呢 ，或者是我哪里用错了

我想实现，监听delete事件，并把这个id拿到，然后再传给后端，这样我就能去更新数据库

**Comment by hizzgdev** [comment]
试试这样：

```javascript
//定义一个方法
function callback(type ,data){
    // 看看修改时控制台里有什么数据，选择合适的type完成你的需求。
    console.log(arguments);
}

// 注册到 jsMind 里
jm.add_event_listener(callback);

```

不过这是一个[实验性功能](https://github.com/hizzgdev/jsmind/blob/master/docs/zh/experimental-features.md) 未来接口可能会有所变化，使用的话下次升级时请做好回归测试。

**Comment by MR1499** [comment]
好的 ，非常感谢，我刚刚试了一下 我禁用了快捷键不允许从前端删除 从后端删除在加载 这就解决这个问题了 非常感谢
